coalahtml/_coalahtml/app/controllers/files.js

```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
     };
     $scope.fileName = $routeParams.fileName || $rootScope.ROOTHOME;
     parseCoalaProject($scope.fileName);
-    $scope.basename = function(path_name){
+
+    $scope.basename = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
       return path_name.split('/').slice(-1)[0];
     };
+    var len = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
+      return path_name.split('/').length;
+    };
```

Comment:
You split path every time, this is also not good.

12:04, пт, 10.06.2016 Tushar <notifications@github.com>:

> In coalahtml/_coalahtml/app/controllers/files.js
> <https://github.com/coala-analyzer/coala-html/pull/52#discussion_r66582483>
> :
>
> >        return path_name.split('/').slice(-1)[0];
> >      };
> > +    var len = function(path_name) {
> > +      path_name = path_name || $rootScope.ROOTHOME;
> > +      return path_name.split('/').length;
> > +    };
>
> I am calculating the same thing (4922d4c
> #diff-b803c22fe9d790db802829435dd58480R53)%5B2
> <https://github.com/coala-analyzer/coala-html/pull/52/commits/4922d4c32e9d92ba2d9764ef3f6a3cdb035d93b9#diff-b803c22fe9d790db802829435dd58480R53)%5B2>
> times]. To make the getNameLink clean, I need to separate the process.
> I agree, the name is ambiguous. I will change the name.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/52/files/5462509acc4ed0d73c39c50f8d08cd53ae35ac77..4922d4c32e9d92ba2d9764ef3f6a3cdb035d93b9#r66582483>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADyQxXVdANhJf9UGFtYG_akffWFwfTdBks5qKSiRgaJpZM4IslPq>
> .
>